genesyscloud/guide/resource_genesyscloud_guide.go

```diff
@@ -0,0 +1,151 @@
+	}
+
+	guide, resp, err := proxy.createGuide(ctx, guideReq)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(ResourceType, fmt.Sprintf("Failed to create guide: %s | error: %s", name, err), resp)
+	}
+
+	d.SetId(*guide.Id)
+	log.Printf("Created guide: %s", *guide.Name)
+	return readGuide(ctx, d, meta)
+}
+
+func readGuide(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getGuideProxy(sdkConfig)
+	cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceGuide(), constants.ConsistencyChecks(), ResourceType)
+
+	log.Printf("Reading Guide: %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		guide, resp, err := proxy.getGuideById(ctx, d.Id())
+		if err != nil {
+			if util.IsStatus404(resp) {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(ResourceType, fmt.Sprintf("Failed to read guide %s | Error: %s", d.Id(), err), resp))
+			}
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(ResourceType, fmt.Sprintf("Failed to read guide %s | Error: %s", d.Id(), err), resp))
+		}
+
+		resourcedata.SetNillableValue(d, "name", guide.Name)
+
+		if guide.Status != nil {
+			_ = d.Set("status", *guide.Status)
+		}
+		if guide.Source != nil {
+			_ = d.Set("source", guide.Source)
+		}
+		if guide.LatestSavedVersion != nil && guide.LatestSavedVersion.Version != nil {
+			_ = d.Set("latest_saved_version", guide.LatestSavedVersion.Version)
+		}
+		if guide.LatestProductionReadyVersion != nil && guide.LatestProductionReadyVersion.Version != nil {
+			_ = d.Set("latest_production_ready_version", guide.LatestProductionReadyVersion.Version)
+		}
+
+		log.Printf("Read Guide: %s", d.Id())
+		return cc.CheckState(d)
+	})
+}
+
+func deleteGuide(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getGuideProxy(sdkConfig)
+
+	log.Printf("Deleting Guide: %s", d.Id())
+
+	job, resp, err := proxy.deleteGuide(ctx, d.Id())
+	if err != nil {
+		if util.IsStatus404(resp) {
+			log.Printf("Guide %s already deleted", d.Id())
+			return nil
+		}
+		return util.BuildAPIDiagnosticError(ResourceType, fmt.Sprintf("Failed to delete guide %s | Error: %s", d.Id(), err), resp)
+	}
+
+	if resp.StatusCode == 202 {
+		log.Printf("Delete Job for Guide: %s started", d.Id())
+	}
+
+	// Large Timeout to allow for delete job to complete before context deadline exceeded
+	return util.WithRetries(ctx, 20*time.Second, func() *retry.RetryError {
+		guide, resp, err := proxy.getGuideById(ctx, d.Id())
+		if err != nil {
+			if util.IsStatus404(resp) {
+				log.Printf("Deleted Guide: %s", d.Id())
+				return nil
+			}
+			return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(ResourceType, fmt.Sprintf("error retrieving guide %s | error: %s", d.Id(), err), resp))
+		}
+
+		if guide != nil {
```

Comment:
Do we need to check if guide is nil here? doesn't seem to make a difference. Also, if it is nil then we go right to the retryable error "guide still exists" which doesn't really make sense. Maybe that error should be `"Unexpected job status: %s"`